packages/react-core/src/components/NumberInput/examples/NumberInput.md

```diff
@@ -400,3 +412,66 @@ class CustomStepNumberInput extends React.Component {
   }
 }
 ```
+
+### Custom increment/decrement and thresholds
+
+```js
+import React from 'react';
+import { NumberInput } from '@patternfly/react-core';
+
+class CustomStepNumberInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: 90,
+      step: 3
+    };
+    this.minValue = 90;
+    this.maxValue = 100;
+
+    this.normalizeBetween = (value, min, max) => {
+      let result = value;
+      if (min !== undefined && max !== undefined) {
+        result = Math.max(Math.min(value, max), min);
+      } else if (max === undefined && value <= min) {
+        result = min;
+      } else if (min === undefined && value >= max) {
+        result = max;
+      }
+      return result;
+    };
```

Comment:
Nit: I think getting rid of the mutation and logically unneeded undefined checks in the second and third if clean this up a lot.

```suggestion
    this.normalizeBetween = (value, min, max) => {
      if (min !== undefined && max !== undefined) {
        return Math.max(Math.min(value, max), min);
      } else if (value <= min) { 
        return min;
      } else if (value >= max) { 
        return max;
      }
      return value;
    };
```
```